gunicorn/arbiter.py

```diff
@@ -482,9 +488,9 @@ def manage_workers(self):
             (pid, _) = workers.pop(0)
             self.kill_worker(pid, signal.SIGTERM)
 
-        self.log.debug("{0} workers".format(len(workers)),
+        self.log.debug("{0} workers".format(len(self.WORKERS)),
                        extra={"metric": "gunicorn.workers",
-                              "value": len(workers),
+                              "value": len(self.WORKERS),
```

Comment:
Before, `workers` and `self.WORKERS` could be out of sync here if workers were killed, because they would be reaped asynchronously. Because both their insertion (via `spawn_worker`) and removal (via `kill_worker`) are now synchronous, `self.WORKERS` is the correct value here.